File: backend/model/src/main/java/org/example/model/entity/token/ResetPasswordToken.java
package org.example.model.entity.token;

import lombok.NoArgsConstructor;
import org.example.constant.PathnameConstants;

@NoArgsConstructor
public class ResetPasswordToken extends ConfirmationToken {
    @Override
    public void prePersist() {
        super.prePersist();
        this.pathname = PathnameConstants.RESET_PASSWORD;
    }
}

File: laptop-store-backend/model/src/main/kotlin/org/example/model/entity/token/ResetPasswordToken.kt
package org.example.model.entity.token

import org.example.constant.PathnameConstants
import org.example.constant.TokenConstants
import javax.persistence.DiscriminatorValue
import javax.persistence.Entity
import javax.persistence.PrePersist

@Entity
@DiscriminatorValue(TokenConstants.RESET_PASSWORD)
class ResetPasswordToken : ConfirmationToken() {
    @PrePersist
    override fun prePersist() {
        super.prePersist()
        this.pathname = PathnameConstants.RESET_PASSWORD
    }
}